apl-core/src/main/java/com/apollocurrency/aplwallet/apl/exchange/service/DexSmartContractService.java

```diff
@@ -261,13 +255,13 @@ public SwapDataInfo getSwapData(Credentials credentials, byte[] secretHash) thro
     }
 
     public String getHashForAtomicSwapTransaction(long orderId) throws NoSuchElementException {
-        DexContract dexContract = new DexContractImpl(smartContractAddress, web3j, Credentials.create(ACCOUNT_TO_READ_DATA), null);
+        DexContract dexContract = new DexContractImpl(smartContractAddress, dexBeanProducer.web3j(), Credentials.create(ACCOUNT_TO_READ_DATA), null);
```

Comment:
I'd suggest you to save web3j to field to not block other methods, which also utilize such call since its blocking (synchronized)

Reply:
I've improved getting an instants, so it's not blocking operation now.